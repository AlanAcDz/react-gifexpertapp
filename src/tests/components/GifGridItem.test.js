import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import '@testing-library/jest-dom';

describe('<GifGridItem /> tests', () => {
    const title = 'Un Titulo';
    const url = 'https://test.com';
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);
    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should have a paragraph with title text', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    test('should have an image with src url and alt text', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    });
    test('should have a div with fadeIn class', () => {
       const div = wrapper.find('div');
       expect(div.hasClass('animate__fadeIn')).toBeTruthy();
    });
});
