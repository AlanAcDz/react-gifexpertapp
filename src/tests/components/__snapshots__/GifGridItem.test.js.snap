// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> tests should render correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://test.com"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
